add first commit

Matvey2283 opened the pull request.

Matvey2283 committed:
add first commit

robotomize commented on .vscode/launch.json:
```diff
@@ -0,0 +1,7 @@
+    // Наведите указатель мыши, чтобы просмотреть описания существующих атрибутов.
+    // Для получения дополнительной информации посетите: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": []
```
Лишние файлы, не относящиеся к проекту нужно игнорировать через файлик .gitignore

robotomize commented on m.go:
```diff
@@ -0,0 +1,210 @@
+package main
```
В данном случае имя файла лучше оставлять как main.go

robotomize commented on m.go:
```diff
@@ -0,0 +1,210 @@
+// Calories возвращает количество потраченных килокалорий при ходьбе.
+func (w Walking) Calories() float64 {
+	speedMetersPerSec := w.meanSpeed() * KmHInMsec
+	return ((CaloriesWeightMultiplier*w.Weight + (speedMetersPerSec*speedMetersPerSec)/float64(w.Height)*CaloriesSpeedHeightMultiplier*w.Weight) * w.Duration.Hours() * MinInHours)
```
В данном коде небольшая ошибка, рост нужно перевести в сантиметры, для этого воспользуйтесь соответствующей константой. Сам рост также уже в float64 поэтому дополнительного его можно не переводить в float64

robotomize commented on m.go:
```diff
@@ -0,0 +1,210 @@
+)
+
+// Swimming структура, описывающая тренировку Плавание
+type Swimming struct {
```
Забыли метод TrainingInfo

robotomize commented on m.go:
```diff
@@ -0,0 +1,210 @@
+
+// TrainingInfo возвращает информацию о тренировке бег.
+func (r Running) TrainingInfo() InfoMessage {
+	return InfoMessage{
```
В данном случае мы можем воспользоваться методом встроенной структуры TrainingInfo

Matvey2283 committed:
Update m.go

Изменил формулы для нахождения ср. скорости при плавании
Изменил формулу для нахождения потраченных килокалорий при плавании и беге

robotomize commented on m.go:
```diff
@@ -118,12 +123,20 @@ type Walking struct {
-	speedMetersPerSec := w.meanSpeed() * KmHInMsec
-	return ((CaloriesWeightMultiplier*w.Weight + (speedMetersPerSec*speedMetersPerSec)/float64(w.Height)*CaloriesSpeedHeightMultiplier*w.Weight) * w.Duration.Hours() * MinInHours)
+    // вставьте ваш код ниже
+    return ((CaloriesWeightMultiplier * w.Weight) + (math.Pow(w.meanSpeed(), 2) / w.Height * CaloriesSpeedHeightMultiplier * w.Weight)) * w.Duration.Hours() * MinInHours
```
Тут у вас 2 ошибки в формуле
- Нужно перевести рост из сантиметров в метры, воспользуйтесь для этого определенной константой в коде или используйте свою 100
- Скорость нужно переветси в м/c для этого воспользуйтесь  соответствующей константой

robotomize commented on m.go:
```diff
@@ -149,20 +162,39 @@ type Swimming struct {
+// длина_гребка * количество_гребков / м_в_км / продолжительность_тренировки
 func (s Swimming) meanSpeed() float64 {
-	return float64(s.LengthPool) * float64(s.CountPool) / MInKm / s.Duration.Hours()
+	return float64(s.LenStep * float64(s.Action)) / MInKm / s.Duration.Hours()
```
Нужно добавить проверку Duration на 0, так как у нас есть потенциально опасная операция

robotomize reviewed (changes requested):
Хорошо, но нужно поправить пару моментов

Matvey2283 committed:
Update m.go

Изменил правки для файла m.go

Matvey2283 committed:
Rename .DS_Store to .DS_Store.gitignore

robotomize commented on m.go:
```diff
@@ -0,0 +1,261 @@
+
+// meanSpeed возвращает среднюю скорость бега или ходьбы.
+func (t Training) meanSpeed() float64 {
+	return t.distance() / t.Duration.Hours()
```
В этом коде также нужно не забывать проверять на 0

robotomize commented on m.go:
```diff
@@ -0,0 +1,261 @@
+// ((18 * средняя_скорость_в_км/ч + 1.79) * вес_спортсмена_в_кг / м_в_км * время_тренировки_в_часах * мин_в_часе)
+// Это переопределенный метод Calories() из Training.
+func (r Running) Calories() float64 {
+	return ((CaloriesMeanSpeedMultiplier*r.meanSpeed() + CaloriesMeanSpeedShift) * r.Weight / MInKm * r.Duration.Hours() * MinInHours)
```
Тут внешние скобки в целом можно опустить

robotomize reviewed (approved):
Отличная работа над ошибками. Принято! Во втором методе расчета скорости также нужно проверять на 0, но это уже самостоятельно поправьте.
Насчет гитигнор прочитайте, может быть [эта ссылка](https://www.atlassian.com/ru/git/tutorials/saving-changes/gitignore) вам поможет
.gitignore это отдельный файл куда вы заносите регулярные выражения для игнорирования гитом. Если вы уже в гит добавили файлик, то он не будет заигнорирован, его надо из гит удалить будет. Желательно в новых проектах создавать файлик .gitignore туда вносить правила игнорирования, например файлы .DS_Store или папку .vscode/
